scripts/o3de/o3de/android_support.py

```diff
@@ -0,0 +1,2116 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+import argparse
+import configparser
+import datetime
+import fnmatch
+import imghdr
+import json
+import logging
+import os
+import re
+import platform
+import shutil
+import stat
+import string
+import subprocess
+
+from enum import Enum
+from o3de import command_utils, manifest, utils
+from packaging.version import Version
+from pathlib import Path
+from typing import Dict, List, Tuple
+
+logging.basicConfig(format=utils.LOG_FORMAT)
+logger = logging.getLogger('o3de.android')
+
+DEFAULT_READ_ENCODING = 'utf-8'    # The default encoding to use when reading from a text file or stream
+DEFAULT_WRITE_ENCODING = 'ascii'
+ENCODING_ERROR_HANDLINGS = 'ignore'     # What to do if we encounter any encoding errors
+
+# Set platform specific values and extensions
+if platform.system() == 'Windows':
+    EXE_EXTENSION = '.exe'
+    O3DE_SCRIPT_EXTENSION = '.bat'
+    SDKMANAGER_EXTENSION = '.bat'
+    GRADLE_EXTENSION = '.bat'
+    DEFAULT_ANDROID_SDK_PATH = f"{os.getenv('LOCALAPPDATA')}\\Android\\Sdk"
+    PYTHON_SCRIPT = 'python.cmd'
+elif platform.system() == 'Darwin':
+    EXE_EXTENSION = ''
+    O3DE_SCRIPT_EXTENSION = '.sh'
+    SDKMANAGER_EXTENSION = ''
+    GRADLE_EXTENSION = '.sh'
+    DEFAULT_ANDROID_SDK_PATH = f"{os.getenv('HOME')}/Library/Android/Sdk"
+    PYTHON_SCRIPT = 'python.sh'
+elif platform.system() == 'Linux':
+    EXE_EXTENSION = ''
+    O3DE_SCRIPT_EXTENSION = '.sh'
+    SDKMANAGER_EXTENSION = ''
+    GRADLE_EXTENSION = '.sh'
+    DEFAULT_ANDROID_SDK_PATH = f"{os.getenv('HOME')}/Android/Sdk"
+    PYTHON_SCRIPT = 'python.sh'
+else:
+    assert False, f"Unknown platform {platform.system()}"
+
+ASSET_MODE_LOOSE = 'LOOSE'
+ASSET_MODE_PAK = 'PAK'
+
+ASSET_MODES = [ASSET_MODE_LOOSE, ASSET_MODE_PAK]
+
+BUILD_CONFIGURATIONS = ['Debug', 'Profile', 'Release']
+ANDROID_ARCH = 'arm64-v8a'
+
+DEFAULT_ANDROID_GRADLE_PLUGIN_VERSION = '8.1.0'
+ANDROID_RESOLUTION_SETTINGS = ('mdpi', 'hdpi', 'xhdpi', 'xxhdpi', 'xxxhdpi')
+
+# Values to set for the android configuration handler
+ANDROID_SETTINGS_FILE = '.command_settings'
```

Comment:
I think it is better to place settings files like this underneath a folder instead of a file as it allows the entire folder to be .gitignored and if for some reason more files need to be added for android configuration settings, the scripts don't have to be update to account for that case.

Reply:
This was only one file so I'm reluctant to do that for now. This settings file uses `ConfigParser`'s sections so if there is need to add more specific sections that are not android related.